app/controllers/departments/list.js

```diff
@@ -1,8 +1,55 @@
 import { db } from 'app/utils/db.server';
+import { getPagination } from 'app/controllers/users/list';
 
-const listDepartments = async () => {
-  const departments = await db.Departments.findMany({});
-  return departments;
+const buildWhere = (value) => {
+  if (!value) return {};
+  return {
+    name: {
+      contains: value,
+    },
+  };
+};
+
+const calLimit = (allPages, count, limit) => {
+  if (allPages) return count;
+  return limit;
+};
+
+const calOffset = (allPages, offset, limit, totalPages) => {
+  if (allPages) return 0;
+  if ((offset / limit) > totalPages) return 0;
+
+  return offset;
+};
+
+const listDepartments = async (params) => {
+  let page; let size; let search; let allPages;
```

Comment:
I think you can declare multiple variables like this:

```suggestion
  let page, size, search, allPages;
```
So you don't have to repeat multiple times the keyword _let_.

References:
https://linuxhint.com/javascript-declare-multiple-variables/#3